### PR TITLE
[openshift-saas-deploy] initial support for helm charts

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2250

depends on https://github.com/app-sre/qontract-schemas/pull/676

with this PR we intend to enable deploying helm charts.

example MR: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/109504

### CODE REVIEW

File: reconcile/utils/helm.py [hunk 1]
```diff
@@ -47,3 +52,12 @@ def template(values: Mapping[str, Any]) -> Mapping[str, Any]:
...
 
     return yaml.safe_load(result.stdout.decode())
+
+
+def template_from_repo(
+    url: str, path: str, name: str, values: Mapping[str, Any]
+) -> Mapping[str, Any]:
+    with tempfile.TemporaryDirectory() as wd:
+        git.clone(url, wd)
+        print(wd)
```
Comment (jfchevrette) [inline reconcile/utils/helm.py:62]: debug print can be removed?

Comment (maorfr) [reply reconcile/utils/helm.py]: https://github.com/app-sre/qontract-reconcile/pull/4472/commits/a667bb7c28da6ea41529a96d4325c5d252262b97

File: reconcile/utils/helm.py [hunk 2]
```diff
@@ -46,4 +51,26 @@ def template(values: Mapping[str, Any]) -> Mapping[str, Any]:
...
+
+def template_all(
+    url: str,
+    path: str,
+    name: str,
+    values: Mapping[str, Any],
+    ssl_verify: bool = True,
+) -> Iterable[Mapping[str, Any]]:
+    with tempfile.TemporaryDirectory() as wd:
+        git.clone(url, wd, verify=ssl_verify)
```
Comment (hemslo) [inline reconcile/utils/helm.py:73]: since we just need files, shallow clone is enough (depth=1)

**Review by hemslo — APPROVED** [review]
LGTM! Need to doc this is helm chart template support, not standard helm chart repo support.

File: reconcile/utils/helm.py (hunk 2 above)
Comment (maorfr) [reply reconcile/utils/helm.py]: https://github.com/app-sre/qontract-reconcile/pull/4472/commits/b0685754fe50c1edcb9c5e83bc47d64c7a51cc48

**Pull request merged**